Assets/Scripts/Auction/AuctionDriver.cs

```diff
@@ -156,6 +182,36 @@ private bool TryPlaceBid(PlayerManager player, int slot)
         return true;
     }
 
+    private IEnumerator AnimateGunConstruction(PlayerManager playerManager, RectTransform parent)
+    {
+        yield return new WaitForSeconds(1);
+        GameObject body = Instantiate(playerManager.identity.Body.augment, parent);
+        body.LeanScale(new Vector3 (gunConstructionScale, gunConstructionScale, gunConstructionScale), 0.5f);
+        body.LeanRotateY(90f, 2f);
+        GunBody gunBody = body.GetComponent<GunBody>();
+
+        yield return new WaitForSeconds(1);
+        GameObject barrel = Instantiate(playerManager.identity.Barrel.augment, gunBody.attachmentSite.position, gunBody.attachmentSite.rotation, parent);
+        barrel.LeanScale(new Vector3(gunConstructionScale, gunConstructionScale, gunConstructionScale), 0.5f);
+        barrel.LeanRotateY(90f, 2f);
+        GunBarrel gunBarrel = barrel.GetComponent<GunBarrel>();
+
+        if (playerManager.identity.Extension)
+        {
+            yield return new WaitForSeconds(1);
+            GameObject extension = Instantiate(playerManager.identity.Extension.augment, gunBarrel.attachmentPoints[0].position, gunBarrel.attachmentPoints[0].rotation, parent);
+            GunExtension gunExtension = extension.GetComponent<GunExtension>();
+            var outputs = new List<Transform>();
+            outputs.AddRange(gunExtension.outputs);
+            outputs.AddRange(gunExtension.AttachToTransforms(gunBarrel.attachmentPoints));
+            extension.LeanScale(new Vector3(gunConstructionScale, gunConstructionScale, gunConstructionScale), 0.5f);
+            extension.LeanRotateY(90f, 2f);
```

Comment:
You're duplicating the `LeanScale` and `LeanRotateY` calls here, perhaps they should be put in their own method?